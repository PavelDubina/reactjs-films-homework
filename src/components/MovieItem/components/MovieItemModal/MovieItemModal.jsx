/*
 * Copyright Â© 2020 EPAM Systems, Inc. All Rights Reserved. All information contained herein is, and remains the
 * property of EPAM Systems, Inc. and/or its suppliers and is protected by international intellectual
 * property law. Dissemination of this information or reproduction of this material is strictly forbidden,
 * unless prior written permission is obtained from EPAM Systems, Inc
 */

import React from 'react'
import PropTypes from 'prop-types'
import { faTimesCircle } from '@fortawesome/free-regular-svg-icons'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { useDispatch } from 'react-redux'
import { fetchTrailerData } from '../../../../redux/actions/fetchTrailer'
import { Info } from '../../../common/Info/Info'
import { Button } from '../../../common/Button/Button'
import styles from './MovieItemModal.scss'

export const MovieItemModal = ({ genreIds, id, overview, title, vote, onModal }) => {
  const dispatch = useDispatch()
  return (
    <div className={styles.modal}>
      <div className={styles.modal__window}>
        <FontAwesomeIcon onClick={onModal} className={styles.modal__cross} icon={faTimesCircle} />
        <Info type="info-modal" genreIds={genreIds} title={title} vote={vote} />
        <div className={styles.modal__description}>{overview}</div>
        <div className={styles.modal__button}>
          <Button
            onClick={() => {
              dispatch(fetchTrailerData(id))
              onModal()
            }}
            type="primary"
          >
            Watch Now
          </Button>
        </div>
      </div>
    </div>
  )
}

MovieItemModal.propTypes = {
  onModal: PropTypes.func.isRequired,
  vote: PropTypes.number.isRequired,
  genreIds: PropTypes.arrayOf(PropTypes.number).isRequired,
  overview: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  id: PropTypes.number.isRequired,
}
