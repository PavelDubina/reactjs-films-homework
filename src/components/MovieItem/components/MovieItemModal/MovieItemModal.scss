/*
 * Copyright Â© 2020 EPAM Systems, Inc. All Rights Reserved. All information contained herein is, and remains the
 * property of EPAM Systems, Inc. and/or its suppliers and is protected by international intellectual
 * property law. Dissemination of this information or reproduction of this material is strictly forbidden,
 * unless prior written permission is obtained from EPAM Systems, Inc
 */

.modal {
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  padding: 10px;
  background-color: rgba(0, 0, 0, 0.8);
  border-radius: 10px;
  z-index: 50;

  &__window {
    position: relative;
    padding: 10px;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
  }

  &__cross {
    position: absolute;
    top: 0;
    left: 0;
    color: #fff;
    font-size: 24px;
    cursor: pointer;
  }

  &__description {
    margin: 10px 0;
    color: #fff;
    height: 200px;
    overflow-y: auto;
  }

  &__button {
    align-self: flex-start;
    margin: 0;
    background-color: rgb(24, 183, 236);
    color: #fff;
  }
}

.info {
  flex: 1;
  width: 100%;

  &__about {
    padding: 30px 0 5px 0;
    display: flex;
    justify-content: space-between;

    &_title {
      color: #fff;
      font-size: 24px;
      padding-right: 10px;
      font-weight: bold;
      overflow: hidden;
      white-space: nowrap;
      text-overflow: ellipsis;
    }

    &_vote {
      font-weight: bold;
      color: rgb(24, 183, 236);
      padding: 1px 7px;
      border: 2px solid rgb(24, 183, 236);
      border-radius: 5px;
    }
  }

  &__genres {
    color: rgb(24, 183, 236);
    font-weight: 800;
  }
}
