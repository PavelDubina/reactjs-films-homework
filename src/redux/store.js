/*
 * Copyright Â© 2020 EPAM Systems, Inc. All Rights Reserved. All information contained herein is, and remains the
 * property of EPAM Systems, Inc. and/or its suppliers and is protected by international intellectual
 * property law. Dissemination of this information or reproduction of this material is strictly forbidden,
 * unless prior written permission is obtained from EPAM Systems, Inc
 */

import { createStore, applyMiddleware } from 'redux'
import thunk from 'redux-thunk'
import { composeWithDevTools } from 'redux-devtools-extension'
import rootReducer from './reducers/rootReducer'

const store = createStore(rootReducer, composeWithDevTools(applyMiddleware(thunk)))

export default store
