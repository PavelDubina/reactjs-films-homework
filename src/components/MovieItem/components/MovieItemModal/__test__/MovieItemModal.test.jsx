/*
 * Copyright Â© 2020 EPAM Systems, Inc. All Rights Reserved. All information contained herein is, and remains the
 * property of EPAM Systems, Inc. and/or its suppliers and is protected by international intellectual
 * property law. Dissemination of this information or reproduction of this material is strictly forbidden,
 * unless prior written permission is obtained from EPAM Systems, Inc
 */

import React from 'react'
import { create, act } from 'react-test-renderer'
import { Provider } from 'react-redux'
import store from '../../../../../redux/store'
import { MovieItemModal } from '../MovieItemModal'

describe('MovieItemModal', () => {
  store.dispatch = jest.fn()

  const component = create(
    <Provider store={store}>
      <MovieItemModal genreIds={[28, 12, 16]} id={1} overview="some text" title="title" vote={5} onModal={() => {}} />
    </Provider>,
  )

  it('Should render and match snapshot', () => {
    expect(component.toJSON()).toMatchSnapshot()
  })
  it('should dispatch an action on Button click', () => {
    act(() => {
      component.root.findByType('button').props.onClick()
    })
    expect(store.dispatch).toHaveBeenCalledTimes(1)
  })
})
