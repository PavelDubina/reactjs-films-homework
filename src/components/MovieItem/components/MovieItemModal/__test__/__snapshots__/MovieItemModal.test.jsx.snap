// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieItemModal Should render and match snapshot 1`] = `
<div
  className="modal"
>
  <div
    className="modal__window"
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-times-circle fa-w-16 modal__cross"
      data-icon="times-circle"
      data-prefix="far"
      focusable="false"
      onClick={[Function]}
      role="img"
      style={Object {}}
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm0 448c-110.5 0-200-89.5-200-200S145.5 56 256 56s200 89.5 200 200-89.5 200-200 200zm101.8-262.2L295.6 256l62.2 62.2c4.7 4.7 4.7 12.3 0 17l-22.6 22.6c-4.7 4.7-12.3 4.7-17 0L256 295.6l-62.2 62.2c-4.7 4.7-12.3 4.7-17 0l-22.6-22.6c-4.7-4.7-4.7-12.3 0-17l62.2-62.2-62.2-62.2c-4.7-4.7-4.7-12.3 0-17l22.6-22.6c4.7-4.7 12.3-4.7 17 0l62.2 62.2 62.2-62.2c4.7-4.7 12.3-4.7 17 0l22.6 22.6c4.7 4.7 4.7 12.3 0 17z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
    <div
      className="info_modal"
    >
      <div
        className="info_modal__about"
      >
        <div
          className="info_modal__about_title"
        >
          title
        </div>
        <div
          className="vote"
        >
          5
        </div>
      </div>
      <div
        className="info_modal__genres"
      >
        
      </div>
    </div>
    <div
      className="modal__description"
    >
      some text
    </div>
    <div
      className="modal__button"
    >
      <button
        className="button primary"
        onClick={[Function]}
      >
        Watch Now
      </button>
    </div>
  </div>
</div>
`;
